### PR TITLE
ref: update seat usage to assign users to seats

### PR DESCRIPTION
**Pull request by chdsbd**
Previously if an account went over their seat quantity they would be considered blocked. Now only users in excess of an account's subscription quantity will be blocked.

Basically individual users now occupy seats. If they occupy a seat before an account's seats are consumed, they'll be able to use Kodiak even as excess users are blocked.

This behavior is better because it allows organizations to user the seats they purchased even when they have more users than purchased seats.

### CODE REVIEW

File: web_ui/src/components/UsageBillingPage.tsx [hunk 1]
```diff
@@ -982,7 +993,9 @@ function UsageBillingPageInner(props: IUsageBillingPageInnerProps) {
                     {u.name}
                   </td>
                   <td>{u.interactions}</td>
+                  <td>{u.firstActiveDate}</td>
```
Comment (sbdchd) [inline web_ui/src/components/UsageBillingPage.tsx:996]: when firstActiveDate is undefined what does this show / what should it show?

File: web_api/core/views.py [hunk 2]
```diff
@@ -80,21 +80,38 @@ def usage_billing(request: HttpRequest, team_id: str) -> HttpResponse:
...
+    )
+    # we assign seats to users in order of first active for the last 30 days.
+    allowed_user_ids = set(
+        (
+            user.github_id
+            for user in sorted(active_users, key=lambda x: x.first_active_at)[
+                :subscription_quantity
+            ]
+        )
+    )
```
Comment (sbdchd) [inline web_api/core/views.py:95]:
```suggestion
    allowed_user_ids = {
            user.github_id
            for user in sorted(active_users, key=lambda x: x.first_active_at)[
                :subscription_quantity
            ]
    }
```

File: web_api/core/models.py [hunk 3]
```diff
@@ -691,7 +716,9 @@ def get_active_users_in_last_30_days(account: Account) -> List[ActiveUser]:
                 AND b.is_private_repository = TRUE
                 AND a.github_installation_id = %s
             GROUP BY
-                a.github_user_id;
+                a.github_user_id
+            ORDER BY
```
Comment (sbdchd) [inline web_api/core/models.py:720]: are there tests for this change?

File: web_ui/src/components/UsageBillingPage.tsx (hunk 1 above)
Comment (chdsbd) [reply web_ui/src/components/UsageBillingPage.tsx:996]: Here's what it looks like when those are undefined. I think this is okay.
image

File: web_api/core/models.py (hunk 3 above)
Comment (chdsbd) [reply web_api/core/models.py:720]: fixed

File: web_api/core/views.py (hunk 2 above)
Comment (chdsbd) [reply web_api/core/views.py]: fixed

**Pull request merged**